src/peripherals/adc.ts

```diff
@@ -186,8 +186,7 @@ export class RPADC extends BasePeripheral implements Peripheral {
         value &= 0xfff; // 12 bits
         if (this.fcs & FCS_SHIFT) {
           value >>= 4;
-        }
-        if (this.fcs & FCS_ERR) {
+        } else if (error && this.fcs & FCS_ERR) {
```

Comment:
Thanks! Adding `error &&` to the if condition makes sense to me, but what the motivation behind the `else`?

Reply:
The shift flag makes sure to put just 1 byte samples into the queue. Without the else the samples were basically 2 bytes again after the shift (because of the added 15th bit sometimes). We could move the error block above the shift block and could mask (by 0xff) instead of shifting ... both makes sense to me but I didn't want to change more than necessary :)

Reply:
Looking at the datasheet, it doesn't see like the error bit is affected by shiting:

![image](https://user-images.githubusercontent.com/892318/213873731-474b814b-dc31-4bd5-a29d-f955568add58.png)

(see the 2nd line of the `ERR` bit description)

Reply:
I didn't see that. Nice to see some more information about the error flag! Therefore you're right: Lets get rid of the `else` and merge it?

Reply:
Yes!

Reply:
Can you please amend the PR to remove the else?

Reply:
Done, thank you :)